Windows/MainWindow.GridControls.cs

```diff
@@ -135,24 +189,38 @@ public partial class MainWindow : Window {
             if (Keyboard.FocusedElement is TextBox || !editorIsLoaded) {
                 return;
             }
-            Point mousePos = e.GetPosition(EditorGrid);
-            editorDragSelectStart = mousePos;
-            gridController.GridMouseDown(mousePos);
-            editorMouseDown = true;
+            if (editorIsHoldScrolling) { //Moving with a released wheel and pressing a button
+                scrollEditor_CancelHoldScrolling();
+            } else {
+                Point mousePos = e.GetPosition(EditorGrid);
+                editorDragSelectStart = mousePos;
+                gridController.GridMouseDown(mousePos);
+                editorMouseDown = true;
+            }
         }
         private void scrollEditor_PreviewMouseLeftButtonUp(object sender, MouseButtonEventArgs e) {
             if (!editorIsLoaded) {
                 return;
             }
-            Point mousePos = e.GetPosition(EditorGrid);
-            gridController.GridMouseUp(mousePos);
-            editorMouseDown = false;
+            if (editorIsHoldScrolling) { //Moving with a released wheel and pressing a button
+                scrollEditor_CancelHoldScrolling();
+            }
+            if (editorDragSelectStart.HasValue || !editorIsHoldScrolling) {
```

Comment:
`scrollEditor_CancelHoldScrolling` will update `editorIsHoldScrolling`, check if this logic behaves as expected.